core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -34,11 +34,34 @@
  * using a {@link StaticTableOperations}. This way no Catalog related calls are needed when reading the table data after
  * deserialization.
  */
-abstract class BaseMetadataTable implements Table, Serializable {
+abstract class BaseMetadataTable implements Table, HasTableOperations, Serializable {
   private final PartitionSpec spec = PartitionSpec.unpartitioned();
   private final SortOrder sortOrder = SortOrder.unsorted();
+  private final TableOperations ops;
+  private final Table table;
+  private final String name;
 
-  abstract Table table();
+  protected BaseMetadataTable(TableOperations ops, Table table, String name) {
+    this.ops = ops;
+    this.table = table;
+    this.name = name;
+  }
+
+  abstract MetadataTableType metadataTableType();
+
+  protected Table table() {
+    return table;
+  }
+
+  @Override
+  public TableOperations operations() {
```

Comment:
I guess instead of exposing the metadata table's operation, here we are actually exposing the original base table's operations, wondering if it's worth adding a comment to explain

Reply:
Here we are exposing whatever ops we get in the metadata table. You are right it is currently the same table ops as in the base table. I think that was done intentionally to keep metadata tables in sync. However, we cannot guarantee this won't change in the future. In `BaseMetadataTable`, we don't really know what ops we got.